### ISSUE TITLE
Unhandled Exception: NUnit.Engine.NUnitEngineException: Remote test agent exited with non-zero exit code -1073741571

### ISSUE THREAD

**Issue by bhavnajoshi1989** [nunit/nunit#3129]
I am doing "nunit3-console FirstTest.dll" and i am getting error, please tell me what needs to be done to resolve this issue.

my setup is : 1) NUnit-2.6.0.12051
2) NUnit.Console-3.9.0

My Code: 

```C#
       [TestCase]
       [WebMethod(EnableSession = true)]
       int sum(int a, int b) {
            return 3;
        }

       public void DoTest()
        { 
           getStateList("closem", "-1", "1", "1");   
        }
```

Note: **getStateList** returns **stream**.
output/Error: Unhandled Exception: NUnit.Engine.NUnitEngineException: Remote test agent exited with non-zero exit code -1073741571
   at NUnit.Engine.Services.TestAgency.OnAgentExit(Object sender, EventArgs e)
   at System.Diagnostics.Process.OnExited()
   at System.Diagnostics.Process.RaiseOnExited()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading._ThreadPoolWaitOrTimerCallback.PerformWaitOrTimerCallback(Object state, Boolean timedOut)

Errors, Failures and Warnings
1) Error :
System.Net.Sockets.SocketException : An existing connection was forcibly closed by the remote host
--SocketException
An existing connection was forcibly closed by the remote host

Server stack trace:
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Runtime.Remoting.Channels.SocketStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadFromSocket(Byte[] buffer, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadAndMatchFourBytes(Byte[] buffer)
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadAndMatchPreamble()
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadVersionAndOperation(UInt16& operation)
   at System.Runtime.Remoting.Channels.Tcp.TcpClientSocketHandler.ReadHeaders()
   at System.Runtime.Remoting.Channels.Tcp.TcpClientTransportSink.ProcessMessage(IMessage msg, ITransportHeaders requestHeaders, Stream requestStream, ITransportHeaders& responseHeaders, Stream& responseStream)
   at System.Runtime.Remoting.Channels.BinaryClientFormatterSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]:
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at NUnit.Engine.ITestEngineRunner.Run(ITestEventListener listener, TestFilter filter)
   at NUnit.Engine.Runners.ProcessRunner.RunTests(ITestEventListener listener, TestFilter filter)

Test Run Summary

**Comment by CharliePoole** [comment]
The stack info merely tells us that the test agent crashed. We would have to analyze what caused it to crash in order to know more.

However, there are a few other indicators in the information you included, which lead to questions or suggestions...

1. Based on the name of your assembly, you are probably just getting started with NUnit. Is there a reason you are using such an old version of the framework? If you are merely learning for yourself, I suggest you use the latest (3.11) framework.  If your company uses NUnit V2, I suggest they upgrade. If they can't use NUnit 3 for some reason, it would be better to use NUnit 2.6.4 or higher.

2. You don't indicate that you have installed NUnit V2 framework driver. Without it, the NUnit 3 console can't run NUnit V2 tests. I actually suspect you do have it, however, because you would have a different error message in case you had not.

3. In your code (which I edited to use fences so it displays better in GitHub) you have a `[TestCase]` which returns a value, but there is no indication to NUnit that you expect a value. This is an error in all versions of NUnit. The only mystery is why the error is not being reported properly. Use a void method for tests unless you use `[TestCase]` with an expected Value parameter.

4. You use the attribute `[WebMethod]` but your code appears to have nothingto do with the web. This isn't an NUnit attribute, of course, so it is not causing the problem but it leads me to wonder if you are expecting that attribute to do something in the test.

5. Also in your code there is a method `DoTest()` which is never called. What is its purpose?

In summary, there are lots of things strange in you test but none of them explain the crash unless it's the fact that you have a very old version of NUnit. But I would expect a clearer message in that case.

If you can provide more info, one of us may be able to help.

**Comment by rprouse** [comment]
I am closing since we have not heard back from you. If you have more information, add it to this issue and we will reopen.

**Comment by veeroo** [comment]
We are struggling with the same issue. Is there any way to investigate it on our side? We tried upgrading from 3.8 to 3.10, but still getting same error.

**Comment by veeroo** [comment]
It seems runner explodes when Stack Overflow exception is thrown in our code

**Comment by shoaibshaikh7** [comment]
We are getting the same error while running the test cases with 3.10.0. It would be very helpful to get a fix for the same.

**Comment by chammonsvusion** [comment]
if our code has a StackOverflow nunit crashes in this way ...

**Comment by CharliePoole** [comment]
@rprouse Is there any way to prevent further comments on a closed issue, forcing those with "the same problem" to file a new one?

@veeroo @shoaibshaikh7 @chammonsvusion The original problem, closed in 2019 was quite specific, limited to a certain environment with certain versions of NUnit. It got a lot of attention before being closed. Piggybacking on closed issues almost never gets you that level of attention because the developers know that problems encountered by users are rarely "the same."

If this is still an ongoing problem for any of you, I suggest filing an issue about it with all the info that an issue needs in order to get someone to look at it more closely.

Some hints:
 * When a test has a stack overflow, the process running it crashes. Period. Nothing we can do will prevent that. It has not been possible to recover from a stack overflow since .NET 2.0 came out.
 * If that test is running in-process, then NUnit itself will crash as well. Again, there's nothing we can do about that. How you are running is therefore important information to include.
 * If the test is running out-of-process, the runner itself should not crash. If it does, you should report it as a bug. However, it isn't a bug in the NUnit framework but in whatever runner you are using, e.g.: nunit-console, VS adapter, third-party runners.

**Comment by mikkelbu** [comment]
I think it makes sense to lock this issue, especially with Charlie's hints as the final say.
